packages/hydrogen/src/storefront.ts

```diff
@@ -149,26 +148,25 @@ export function createStorefrontClient<TI18n extends I18nBase>({
     );
   }
 
-  const {
-    getPublicTokenHeaders,
-    getPrivateTokenHeaders,
-    getStorefrontApiUrl,
-    getShopifyDomain,
-  } = createStorefrontUtilities(clientOptions);
-
-  const getHeaders = clientOptions.privateStorefrontToken
-    ? getPrivateTokenHeaders
-    : getPublicTokenHeaders;
-
-  const defaultHeaders = getHeaders({contentType: 'json'});
+  const authorizer = createAuthorizer({
+    storeDomain: clientOptions.storeDomain,
+    storefrontAPIVersion: clientOptions.storefrontApiVersion,
+    publicAccessToken: clientOptions.publicStorefrontToken,
+    privateAccessToken: clientOptions.privateStorefrontToken,
+    contentType: clientOptions.contentType,
+    sdkVariant: 'Hydrogen',
+    sdkVariantSource: 'react',
+    sdkVersion: clientOptions.storefrontApiVersion,
+  });
+
+  const defaultHeaders = authorizer.getApiHeaders({
+    tokenType: clientOptions.privateStorefrontToken ? 'private' : 'public',
+    buyerIp,
+  }) as Record<string, string>;
```

Comment:
`defaultHeaders` here is a plain object but its type is `HeadersInit`, which doesn't allow us to add new headers. This cast `as Record<string, string>` does the trick.